### ISSUE TITLE
Increase AWS default VM size to t2.medium, or at least add release note about it

### ISSUE THREAD

**Issue by davidopp** [kubernetes/kubernetes#16122]
With #16071 we're finally setting proper memory limit for Heapster, but it can be as big as 3G on 250-node cluster (which is the scaling limit we're advertising for 1.2).

Currently we use t2.micro default VM size for AWS, which only has 1G of memory. I think our choices (from easiest to hardest for us) are
* add a release note (and update docs/getting-started-guides/aws.md and ./cluster/aws/options.md) telling people what VM size to use depending on what cluster size they use
* increase default VM size to t2.medium in ./cluster/aws/{config-default,config-test}.sh(and update the aforementioned docs)
* change cluster/aws/config-default.sh to dynamically pick VM size depending on cluster size

t2.medium is 4G and t2.small is 2G, which means the sizes we'd want to use based on @gmarek's data and the values @brendandburns used in #16071 are something like
t2.micro: 1-50 nodes
t2.small: 51-150 nodes
t2.medium: 151+ nodes

ref/ #15747

@justinsb @brendandburns

**Comment by paralin** [comment]
You can also install the apt package "swapspace" to ensure that the memory limit is never hit. I recommend this particularly on the master.

**Comment by vishh** [comment]
FYI: Swap will interfere with "Quality of Service" feature.

On Thu, Oct 22, 2015 at 4:28 PM, Christian Stewart <notifications@github.com

**Comment by paralin** [comment]
@vishh is there some way to get Kubernetes to ignore the swap? the purpose of the swapspace daemon is to just expand the available memory a bit when it gets to critical levels so things don't lock up.

**Comment by vishh** [comment]
@paralin: Enabling swap on the master node might not be an issue.

**Comment by davidopp** [comment]
The issue isn't the master node -- Heapster runs on one of the minions. It's unfortunate that we need to increase the size of all minions just to support Heapster, but I don't know any other way that is feasible in the 1.1 timeframe. In the future we could support heterogeneous minions, or allow system pods to run on the master by default, or try to reduce memory consumption of Heapster (we should absolutely due that last one ASAP anyway).

(BTW the above paragraph is a good example of why "minions" is a better term than "nodes"... Try using "nodes" everywhere I wrote "minion"... But I will try to accept that I lost that battle :) )

As for whether we should use t2.medium or m3.medium, I don't know. I just thought t2.medium is the smallest change from what we already have. I'm not sure we actually thought much about this. And confusingly enough, I noticed that in our AWS getting started guide
https://github.com/kubernetes/kubernetes/blob/master/docs/getting-started-guides/aws.md
we tell people to override MINION_SIZE to m3.medium. So this whole issue is actually a non-issue if people follow the getting-started guide. But it still seems like we should not use a known-to-be-insufficient default VM type in the startup script.
